plugin/pkg/admission/defaulttolerationseconds/admission.go

```diff
@@ -82,7 +82,19 @@ func (p *Plugin) Admit(attributes admission.Attributes) (err error) {
 
 	toleratesNodeNotReady := false
 	toleratesNodeUnreachable := false
-	for _, toleration := range tolerations {
+	for i, toleration := range tolerations {
+		// replace alpha toleration key with GA key (remove alpha)
+		if toleration.Key == algorithm.DeprecatedTaintNodeNotReady {
+			toleration.Key = algorithm.TaintNodeNotReady
+			pod.Spec.Tolerations[i] = toleration
+		}
+
+		// replace alpha toleration key with GA key (remove alpha)
+		if toleration.Key == algorithm.DeprecatedTaintNodeUnreachable {
+			toleration.Key = algorithm.TaintNodeUnreachable
+			pod.Spec.Tolerations[i] = toleration
+		}
+
 		if (toleration.Key == algorithm.TaintNodeNotReady || len(toleration.Key) == 0) &&
 			(toleration.Effect == api.TaintEffectNoExecute || len(toleration.Effect) == 0) {
 			toleratesNodeNotReady = true
```

Comment:
this is pre-existing, but I find this confusing.. a toleration with an empty key and empty effect means we tolerate a node not ready taint?

Reply:
A toleration with empty key and empty effect means the Pod tolerates all taints, including not-ready.